src/test/java/core/basesyntax/service/FileServiceImplTest.java

```diff
@@ -0,0 +1,93 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertArrayEquals;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.Fruit;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+public class FileServiceImplTest {
+    private static final String EMPTY_INPUT = "inputEmpty.csv";
+    private static final String WRONG_FORMAT_INPUT = "inputWrongFormat.csv";
+    private static final String GOOD_INPUT = "inputOk.csv";
+    private static final String GOOD_RESULT = "resultOk.csv";
+
+    private static FileServiceImpl service;
+
+    @BeforeAll
+    public static void setUp() {
+        service = new FileServiceImpl();
+    }
+
+    @Test
+    void read_ok() {
+        List<String> expectedRows = new ArrayList<>();
+        expectedRows.add("b,banana,20");
+        expectedRows.add("b,apple,100");
+        expectedRows.add("s,banana,100");
+
+        List<String> fileRows = service.readFile(GOOD_INPUT);
+        assertEquals(expectedRows, fileRows);
+    }
+
+    @Test
+    void read_wrongNameFile_notOk() {
+        assertThrows(NullPointerException.class, () -> service.readFile("blabla"));
+    }
+
+    @Test
+    void read_wrongDataFormat_ok() {
+        List<String> expectedRows = new ArrayList<>();
+        expectedRows.add("b,banana,20");
+        expectedRows.add("b,apple,100");
+        expectedRows.add("s--banana--100");
+        expectedRows.add("p,banana,13");
+
+        List<String> fileRows = service.readFile(WRONG_FORMAT_INPUT);
+        assertEquals(expectedRows, fileRows);
+    }
```

Comment:
if you did not handle the case with the wrong format on the reader level, so why you don't have any logic to handle it on service level?